src/mgr/ActivePyModules.cc

```diff
@@ -702,7 +702,7 @@ PyObject* ActivePyModules::get_counter_python(
       const auto &avg_data = counter_instance.get_data_avg();
       for (const auto &datapoint : avg_data) {
         f.open_array_section("datapoint");
-        f.dump_unsigned("t", datapoint.t.sec());
+        f.dump_unsigned("t", datapoint.t.to_nsec());
```

Comment:
Great catch! And no performance penalty (src/include/utime.h).

I'm just concerned about 2 things (1 has little to do with this PR):
- Does this come from a hi-res monotonic clock? Based on: ["
This clock has the same granularity as lowres_clock, but it is not required to be monotonic and its time points correspond to system time."](http://docs.seastar.io/master/classseastar_1_1lowres__system__clock.html), it doesn't seem so, and it seems [granularity is 10 ms](http://docs.seastar.io/master/lowres__clock_8hh_source.html). That means that it might potentially exhibit negative deltas on Daylight Saving Time, Leap seconds, and NTP clock skews (I've experienced this in the past and it's really hard to catch and problematic). 
- This might have impact (for the better) on all consumers of Prometheus metrics. I think this should increase the accuracy of the metrics (haven't heard about any issue reported on that though).

Reply:
> * Does this come from a hi-res monotonic clock? Based on: ["
>   This clock has the same granularity as lowres_clock, but it is not required to be monotonic and its time points correspond to system time."](http://docs.seastar.io/master/classseastar_1_1lowres__system__clock.html), it doesn't seem so, and it seems [granularity is 10 ms](http://docs.seastar.io/master/lowres__clock_8hh_source.html). That means that it might potentially exhibit negative deltas on Daylight Saving Time, Leap seconds, and NTP clock skews (I've experienced this in the past and it's really hard to catch and problematic).

This timestamp comes from `clock_gettime(CLOCK_REALTIME, ...`. From the man page:
```
This clock is affected by discontinuous jumps in the system time (e.g., if the system administrator manually changes the clock), and by the incremental adjustments performed by adjtime(3) and NTP
```

So yes, the consumers of these timestamps (like the dashboard) should take care of sanitize a sequence of timestamps in order to cope with possible "back in time" timestamps. 

> * This might have impact (for the better) on all consumers of Prometheus metrics. I think this should increase the accuracy of the metrics (haven't heard about any issue reported on that though).

Are you talking about the prometheus mgr module? That module does not use the same perf counter information that we are changing in this PR. They are getting numbers from other sources.

Reply:
> This timestamp comes from `clock_gettime(CLOCK_REALTIME, ...`. From the man page:
> 
> ```
> This clock is affected by discontinuous jumps in the system time (e.g., if the system administrator manually changes the clock), and by the incremental adjustments performed by adjtime(3) and NTP
> ```
> 
> So yes, the consumers of these timestamps (like the dashboard) should take care of sanitize a sequence of timestamps in order to cope with possible "back in time" timestamps.

Ah, ok. I assumed this was using the Seastar layer.

The bad thing of negative time skews is that:
- You get negative rates and very often in spikes (a positive time jump usually ends up in lower-than-real rates).
- Additionally, if you (still) use unsigned integers for deltas or rates, you may have real fun with overflows.

> Are you talking about the prometheus mgr module? That module does not use the same perf counter information that we are changing in this PR. They are getting numbers from other sources.

Prometheus `Module.collect`:
_`MgrModule.get_all_perf_counters`
__`MgrModule.Mgr.get_latest`
___`MgrModule.get_latest_counter` 
____`BaseMgrModule::_ceph_get_latest_counter`
_____`ActivePyModules::get_counter_python`

I just mean that this might impact for the better on other perf counter consuming modules, but sometimes you're prepared for the worse and not for the better :sweat_smile:

Reply:
> Prometheus `Module.collect`:
> _`MgrModule.get_all_perf_counters`
> __`MgrModule.Mgr.get_latest`
> ___`MgrModule.get_latest_counter`
> ____`BaseMgrModule::_ceph_get_latest_counter`
> _____`ActivePyModules::get_counter_python`
>

Thanks, I missed this `get_all_perf_counters` call.
 
> I just mean that this might impact for the better on other perf counter consuming modules, but sometimes you're prepared for the worse and not for the better

Hehe, indeed. I will double-check if there is a problem here.

Reply:
`get_all_perf_counters` drops the timestamps info of the perf counters data points, so we are good here :-)